entry/src/block_component.rs

```diff
@@ -618,6 +618,49 @@ impl BlockComponent {
             }
         }
     }
+
+    /// Parse BlockHeader from bytes
+    pub fn parse_block_header_from_data_payload(data: &[u8]) -> Option<(Slot, Hash)> {
+        // Try to deserialize as BlockComponent
+        let (component, _) = BlockComponent::from_bytes(data).ok()?;
+
+        // If we were able to parse the component, it must be a block marker
+        let marker = component.as_marker()?;
+
+        // Check if it's a BlockMarker with BlockHeader
+        match marker {
+            VersionedBlockMarker::V1(BlockMarkerV1::BlockHeader(header))
+            | VersionedBlockMarker::Current(BlockMarkerV1::BlockHeader(header)) => {
```

Comment:
nit: `VersionBlockMarker::Current(...V1)` looks kind of weird, I would imagine something like `VersionBlockMarker::Current(...Current....)`